### ISSUE TITLE
Support System.Linq.IGrouping for Grouped Collections

### ISSUE THREAD

**Issue by zgramana** [NobsterTheLobster/Xamarin.Forms.GridView#24]
[System.Linq.IGrouping](https://docs.microsoft.com/en-us/dotnet/api/system.linq.igrouping-2?view=netstandard-2.0) provides a useful interface for a grouped item source, where `TKey` can conveniently serve as the group header binding source.

**Comment by NobsterTheLobster** [comment]
Yeah I agree that makes more sense I will incorporate it that when I get a chance.
